Code/Editor/EditorViewportCamera.cpp

```diff
@@ -12,47 +12,138 @@
 #include <AtomToolsFramework/Viewport/ModularViewportCameraControllerRequestBus.h>
 
+#include <EditorViewportSettings.h>
+
 namespace SandboxEditor
 {
+    void SetViewportCameraPosition(const AzFramework::ViewportId viewportId, const AZ::Vector3& position)
+    {
+        const auto& currentCameraTransform = GetViewportCameraTransform(viewportId);
+        SetViewportCameraTransform(
+            viewportId, AZ::Transform::CreateFromQuaternionAndTranslation(currentCameraTransform.GetRotation(), position));
+    }
+
     void SetDefaultViewportCameraPosition(const AZ::Vector3& position)
     {
         auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
         if (auto viewportContext = viewportContextManager->GetDefaultViewportContext())
         {
-            const auto& currentCameraTransform = viewportContext->GetCameraTransform();
-            viewportContext->SetCameraTransform(
-                AZ::Transform::CreateFromQuaternionAndTranslation(currentCameraTransform.GetRotation(), position));
+            SetViewportCameraPosition(viewportContext->GetId(), position);
         }
     }
 
+    void SetViewportCameraRotation(const AzFramework::ViewportId viewportId, const float pitch, const float yaw)
+    {
+        const auto& currentCameraTransform = GetViewportCameraTransform(viewportId);
+        SetViewportCameraTransform(viewportId, TransformFromPositionPitchYaw(currentCameraTransform.GetTranslation(), pitch, yaw));
+    }
+
     void SetDefaultViewportCameraRotation(const float pitch, const float yaw)
     {
         auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
         if (auto viewportContext = viewportContextManager->GetDefaultViewportContext())
         {
-            const auto& currentCameraTransform = viewportContext->GetCameraTransform();
-            viewportContext->SetCameraTransform(
-                AZ::Transform::CreateFromQuaternionAndTranslation(CameraRotation(pitch, yaw), currentCameraTransform.GetTranslation()));
+            SetViewportCameraRotation(viewportContext->GetId(), pitch, yaw);
+        }
+    }
+
+    void SetViewportCameraTransform(const AzFramework::ViewportId viewportId, const AZ::Transform& transform)
+    {
+        AzFramework::ViewportRequestBus::Event(viewportId, &AzFramework::ViewportRequestBus::Events::SetCameraTransform, transform);
+    }
+
+    void SetViewportCameraTransform(
+        const AzFramework::ViewportId viewportId, const AZ::Vector3& position, const float pitch, const float yaw)
+    {
+        SetViewportCameraTransform(viewportId, TransformFromPositionPitchYaw(position, pitch, yaw));
+    }
+
+    void SetDefaultViewportCameraTransform(const AZ::Transform& transform)
+    {
+        auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
+        if (auto viewportContext = viewportContextManager->GetDefaultViewportContext())

```

Comment:
I often see people guard against interface pointers being invalid. Not sure if there's any guidance on whether that's preferred or not.

Reply:
Hmm this has been like this for a while and I don't see any reason why the `ViewportContextRequestsInterface` wouldn't be around. I'd rather leave this as-is until there's a reason to handle it failing (and maybe then returning a success/failure code from the function).